contracts/dao/treasury/tests/integration/treasury.rs

```diff
@@ -64,714 +65,439 @@ fn bonded_adapter_int(
     post_rewards: (Uint128, Uint128, Uint128),
 ) {
     let mut app = App::default();
+    let mut contracts = DeployedContracts::new();
 
     let admin = Addr::unchecked("admin");
     let user = Addr::unchecked("user");
-    //let validator = Addr::unchecked("validator");
-    let admin_auth = init_admin_auth(&mut app, &admin);
-
     let viewing_key = "viewing_key".to_string();
+    let symbol = "TKN";
 
-    let token = snip20::InstantiateMsg {
-        name: "secretSCRT".into(),
-        admin: Some("admin".into()),
-        symbol: "SSCRT".into(),
-        decimals: 6,
-        initial_balances: None,
-        prng_seed: to_binary("").ok().unwrap(),
-        config: Some(snip20::InitConfig {
-            public_total_supply: Some(true),
-            enable_deposit: Some(true),
-            enable_redeem: Some(true),
-            enable_mint: Some(false),
-            enable_burn: Some(false),
-            enable_transfer: Some(true),
-        }),
-        query_auth: None,
-    }
-    .test_init(Snip20::default(), &mut app, admin.clone(), "token", &[])
-    .unwrap();
-
-    let treasury = treasury::InstantiateMsg {
-        admin_auth: admin_auth.clone().into(),
-        viewing_key: viewing_key.clone(),
-        multisig: admin.to_string().clone(),
-    }
-    .test_init(Treasury::default(), &mut app, admin.clone(), "treasury", &[
-    ])
-    .unwrap();
-
-    let manager = treasury_manager::InstantiateMsg {
-        admin_auth: admin_auth.clone().into(),
-        treasury: treasury.address.to_string(),
-        viewing_key: viewing_key.clone(),
-    }
-    .test_init(
-        TreasuryManager::default(),
+    interfaces::dao::init_dao(
         &mut app,
-        admin.clone(),
-        "manager",
-        &[],
-    )
-    .unwrap();
+        &admin.to_string(),
+        &mut contracts,
+        deposit,
+        symbol,
+        vec![AllowanceType::Portion],
+        vec![Cycle::Constant],
+        vec![allowance],
+        vec![Uint128::zero()],
+        vec![vec![alloc_type]],
+        vec![vec![alloc_amount]],
+        vec![vec![Uint128::zero()]],
+        false,
+        false,
+    );
 
-    let adapter = mock_adapter::contract::Config {
-        owner: manager.address.clone(),
-        instant: false,
-        token: token.clone().into(),
-    }
-    .test_init(
-        MockAdapter::default(),
+    // Update treasury
+    interfaces::treasury::update_exec(
         &mut app,
-        admin.clone(),
-        "mock_adapter",
-        &[],
+        &admin.to_string(),
+        &contracts,
+        &symbol.to_string(),
```

Comment:
This should just be symbol
```suggestion
        symbol,
```